soh/src/overlays/actors/ovl_En_Cow/z_en_cow.c

```diff
@@ -209,12 +226,57 @@ void func_809DF730(EnCow* this, GlobalContext* globalCtx) {
     }
 }
 
+bool EnCow_IsSecondCowInTower(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_SOUKO && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -173 || this->actor.world.pos.x + this->actor.world.pos.z == -72);
+}
+
+bool EnCow_IsSecondCowInStable(EnCow* this, GlobalContext* globalCtx) {
+    return 
+        gSaveContext.n64ddFlag && 
+        globalCtx->sceneNum == SCENE_MALON_STABLE && 
+        (this->actor.world.pos.x + this->actor.world.pos.z == -257 || this->actor.world.pos.x + this->actor.world.pos.z == -138);
+}
+
+bool EnCow_IsSecondCowInGrotto(EnCow* this, GlobalContext* globalCtx) {
+    return gSaveContext.n64ddFlag && globalCtx->sceneNum == SCENE_KAKUSIANA && this->actor.world.pos.x + this->actor.world.pos.z == 3194;
+}
+
+bool EnCow_IsSecondCow(EnCow* this, GlobalContext* globalCtx) {
+    return EnCow_IsSecondCowInTower(this, globalCtx) || EnCow_IsSecondCowInStable(this, globalCtx) || EnCow_IsSecondCowInGrotto(this, globalCtx);
+}
+
+GetItemID EnCow_GetItemFromCow(EnCow* this, GlobalContext* globalCtx, bool setFlag) {
+    GetItemID itemId = ITEM_NONE;
+
+    if (Inventory_HasEmptyBottle()) {
+        return GI_MILK;
+    }
+
+    s32 collectableFlag = 0x1A;
+    if (EnCow_IsSecondCow(this, globalCtx)) {
+        collectableFlag = 0x19;
```

Comment:
just want to verify we're not using these flags for anything else in the scenes with cows

Reply:
It might be an idea to save these in a new variable instead, just like rando already has dungeonsDone[]. The collectible flags are a vanilla thing and maintaining a rando specific list of flags might be a little difficult in comparison.

Reply:
I’d prefer the new array approach as well, as this approach had me worried, could you give an example on how this could be done or where it’s been done before?

Reply:
As said above, take a look at how the dungeonsDone array is set/saved/used.

Reply:
Ah right, thanks I’ll check it out

Reply:
If you do go this route, you'll probably need to expand the system I set up for setting pending flags on the player. Should be a simple matter of adding an additional value to the `FlagType` enum, setting that value when the player is getting a cow check, and adding a case for that value to the `Player_SetFlag` function.

Reply:
Okay, so I went down this route and it seems to work temporarily; leaving room and coming back works but reloading the save it's lost, any idea if I'm missing a piece?